3.workload.md

```diff
@@ -0,0 +1,378 @@
+# 3. Workload
+
+This section defines the workload definition schematics.
+
+Workload definitions are provided by the platform (just as traits are) so that users may inspect the platform and
+ learn what workloads are available for use. Workload definitions are not extensible to platform users (only to platform
+  operators). Thus, platform users MUST NOT be allowed to create new workload kinds.
```

Comment:
We need to use terms like `Application Developers` etc in the whole doc instead of "platform users" etc for better consistency.

Another suggestion to make above narrative clearer:

Workload Definition is provided by the platform to enforce and validate the workload specification of given `Component`. It's designed to be a platform level concept and expected to be managed by Infrastructure Operators only. The OAM platform should provide a proper approach (e.g. docs or samples) to help Application Developers define `spec.workload` section in Component instead of exposing Workload Definitions directly to them.

Reply:
> Put otherwise, I agree that we should explain that `WorkloadDefinition` is intended for use by infrastructure operators, but I do not think we should go so far as to say application operators MUST NOT be allowed to create workload kinds.

Agree,  this is copied from the existing spec (https://github.com/oam-dev/spec/blob/master/7.workload_types.md#7-workload-types) so I didn't change it as this PR is just about the schema change.  I didn't know why we put such a strong statement in the begining. We can add a PR to resolve this.

Reply:
I'd suggest we change the narrative in this PR :-)